Update to Rocky Linux 9.3

seunghun1ee opened the pull request:
Update pulp repository versions from gh action "Update Kayobe package repository versions" artifact `pulp-repo-versions.yml`

This was done to test Rocky Linux 9.3 migration test.

Action Link: https://github.com/stackhpc/stackhpc-release-train/actions/runs/7128080191

seunghun1ee committed:
Update pulp repo versions from gh action artifact

Update pulp repository versions from gh action "Update Kayobe package
repository versions" artifact `pulp-repo-versions.yml`

seunghun1ee committed:
Fix Wazuh manager playbook with no custom policies

seunghun1ee committed:
Add new line at the end of file because Tox unhappy

Alex-Welsh commented:
Once you've tested these changes, please add a release note. Let me know if you need a hand with doing so.

seunghun1ee commented:
> Once you've tested these changes, please add a release note. Let me know if you need a hand with doing so.

Will do.

seunghun1ee committed:
Update default stackhpc pulp repo rocky version to 9.3

markgoddard commented on etc/kayobe/pulp-repo-versions.yml:
Do we want to update all of these or just the RL9 repos? It would require quite a lot of testing to update everything.

seunghun1ee replied:
Thanks for pointing that out. No, we just need RL9.2 and RL9.3 repos to be updated. There's new action running to update pulp repo. I will update this file with only 9.2 and 9.3 changes and revert other repos.

seunghun1ee committed:
Revert unnecessary updates and update RL 9.2 and 9.3 repos

seunghun1ee committed:
Update Kolla container image tags to RL 9.3

seunghun1ee committed:
Update pulp repo to include RL kernel-5.14.0-362.13.1.el9_3.x86_64

markgoddard committed:
Merge pull request #844 from stackhpc/yoga-opensearch-pin

use ark repos for R9 opensearch builds

seunghun1ee committed:
Fix Rocky Linux 9 targeting Cephadm Pacific

The upstream Cephadm Pacific repository is no more available and this change prevents targeting the repository when building Rocky Linux 9 environment

seunghun1ee committed:
Enable installing Elrepo9 for Rocky Linux 9.3 upgrade

seunghun1ee committed:
Update Kolla container tags to apply pyroute2 compatibility fix

seunghun1ee commented:
Force push was done to revert the accidental merge with this branch and stackhpc/yoga
Commit 3f92251 was cherry-picked to fix Wazuh manager playbook crashing when there's no custom polices
PR #844 was cherry-picked to fix the issue where Opensearch dashboards cannot get log data from Opensearch after upgrading to Rocky Linux 9.3 Kolla containers.

All Tempest tests are passed.
Octavia test is passed.
Horizon, Opensearch Dashboards, Prometheus Alert Manager, Grafana Dashboards UIs are checked.

Note: This upgrade test was done with Barbican and internal TLS disabled in multinode environment.

seunghun1ee committed:
Merge branch 'stackhpc/yoga' into rocky-linux-9.3-package-update

seunghun1ee committed:
Add releasenote

seunghun1ee committed:
Add empty line at release note

Alex-Welsh commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -13,16 +13,24 @@ kolla_base_distro: "{% raw %}{{ 'centos' if ansible_facts.distribution == 'Rocky
     ubuntu: yoga-20231103T161400
-
-cloudkitty_tag: yoga-20231107T165648
+  cloudkitty:
```
nit: it would be nice to have these in alphabetical order

Alex-Welsh commented on etc/kayobe/dnf.yml:
```diff
@@ -200,7 +200,7 @@ rocky_9_gpg_key: "https://dl.rockylinux.org/pub/rocky/RPM-GPG-KEY-Rocky-9"
 # Whether to create a repo file for ELRepo. This affects RedHat-based
 # systems only.
-dnf_install_elrepo_9: false
+dnf_install_elrepo_9: true
```
Are you sure you want to change the default here?

jovial replied:
Agreed, elrepo is known as a bit of a wild west of packages. Were you installing something specific from there?

seunghun1ee replied:
This change is not related to 9.3 upgrade, sorry. Please see my comment below.

Alex-Welsh commented on etc/kayobe/stackhpc.yml:
```diff
@@ -199,7 +199,7 @@ stackhpc_repo_epel_9_version: "{{ stackhpc_repo_distribution }}"
 # ELRepo 9
 stackhpc_repo_elrepo_9_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/elrepo/elrepo/el9/x86_64/{{ stackhpc_repo_elrepo_9_version }}"
-stackhpc_repo_elrepo_9_version: "{{ stackhpc_repo_distribution }}"
+stackhpc_repo_elrepo_9_version: "20230907T075311"
```
I don't think this is the correct place to override this variable. I can't remember it exactly now but can find it for you if you need

priteau replied:
This was set to `stackhpc_repo_distribution` which by default is `development`, but can be changed to `production`. Why are you changing it to a timestamp?

seunghun1ee replied:
This change is not related to 9.3 upgrade, sorry. Please see my comment below.

Alex-Welsh commented on etc/kayobe/cephadm.yml:
These changes are separate, so it's worth adding a second release note for them

seunghun1ee replied:
Added second release note!

seunghun1ee committed:
Revert unused pulp repo version tags

seunghun1ee committed:
List kayobe_image_tags to be alphabetical

seunghun1ee committed:
Add release note for cephadm fix

seunghun1ee committed:
Revert "Fix Rocky Linux 9 targeting Cephadm Pacific"

This reverts commit 3b30a86592f5096b6171a9d6fcda9acb637494b7.

seunghun1ee committed:
Revert "Enable installing Elrepo9 for Rocky Linux 9.3 upgrade"

This reverts commit 6ffa526fa9bf3e1d80ac0ca42f04d29d9e1f6d00.

seunghun1ee committed:
Remove wrong release note

seunghun1ee commented:
Sorry for the confusion.
I redeployed multinode with Rocky Linux 9.2 to check if the changes for `etc/kayobe/stackhpc.yml` and `etc/kayobe/kolla/globals.yml` about elrepo was necessary.
These changes were made to avoid ceph version problem by using ceph-pacific from test pulp rather than upstream since upstream ceph-pacific is no longer available for Rocky Linux 9.
Because it's not proper fix and it's not really related to 9.3 upgrade itself, I now excluded these changes and the one at `etc/kayobe/cephadm.yml`.
I will re-propose cephadm version fix in separate PR.

Alex-Welsh commented on releasenotes/notes/add-rocky-9.3-repos-454c2f825e088417.yaml:
```diff
@@ -0,0 +1,5 @@
+---
+features:
```
Please also add a second release note that mentions the new container images and repo versions. You can either just add a second file or add a second release note to the same file with this format:
```yaml
---
features:
  - |
    First release note here
  - |
    Second release note here

seunghun1ee replied:
Added them!

Alex-Welsh replied:
The new release note mentions the rocky rebuild, but also needs the new cloudkitty and neutron tags for centos and ubuntu.

seunghun1ee replied:
Since tag `yoga-20231107T165648` (cloudkitty) and `yoga-20231114T125927` (neutron) are used for all 3 OS in `stackhpc/yoga`, the change was to preserve the use of the old tag for Centos and Ubuntu. There's no newly built cloudkitty or neutron for them.

Alex-Welsh replied:
Ah, yes my mistake

seunghun1ee committed:
Add note about new contianer images and repo versions

Alex-Welsh commented:
You might want to rebase on the latest stackhpc/yoga to make sure you are up to date.

Alex-Welsh commented on .github/workflows/stackhpc-build-kayobe-image.yml:
```diff
@@ -5,6 +5,11 @@
 name: Build kayobe image
 
 on:
+  push:
```
Something strange is going on here. I'm not exactly sure why this is showing up as your change

seunghun1ee committed:
Merge branch 'stackhpc/yoga' into rocky-linux-9.3-package-update

Alex-Welsh reviewed (approved):
That looks better. Sorry for leading you the wrong way
